src/Db/Adapter/UnifiedMigrationsTableStorage.php

```diff
@@ -0,0 +1,270 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @license       https://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Migrations\Db\Adapter;
+
+use Cake\Database\Query\SelectQuery;
+use Exception;
+use InvalidArgumentException;
+use Migrations\Db\Table;
+use Migrations\MigrationInterface;
+
+/**
+ * Unified migration table storage.
+ *
+ * Uses a single `cake_migrations` table with a `plugin` column
+ * to track all migrations (app and plugins) in one place.
+ */
+class UnifiedMigrationsTableStorage
+{
+    /**
+     * The table name for unified migrations storage.
+     */
+    public const TABLE_NAME = 'cake_migrations';
+
+    /**
+     * Constructor
+     *
+     * @param \Migrations\Db\Adapter\AbstractAdapter $adapter The database adapter.
+     * @param string|null $plugin The plugin name (null for app migrations).
+     */
+    public function __construct(
+        protected AbstractAdapter $adapter,
+        protected ?string $plugin = null,
+    ) {
+    }
+
+    /**
+     * Gets all the migration versions for the current plugin context.
+     *
+     * @param array<string, string> $orderBy The order by clause.
+     * @return \Cake\Database\Query\SelectQuery
+     */
+    public function getVersions(array $orderBy): SelectQuery
+    {
+        $query = $this->adapter->getSelectBuilder();
+        $query->select('*')
+            ->from(self::TABLE_NAME)
+            ->orderBy($orderBy);
+
+        if ($this->plugin === null) {
+            $query->where(['plugin IS' => null]);
+        } else {
+            $query->where(['plugin' => $this->plugin]);
+        }
```

Comment:
```suggestion
        $query->where(['plugin IS' => $this->plugin]);

```

The `Query` translation/compiler will do the right thing here.